### ISSUE TITLE
Add type parameter in Channel.start_thread()

### ISSUE THREAD

**Issue by AiroPi** [Rapptz/discord.py#7261]
### Summary

Adding a `type` parameter to the `start_thread` function, because the API allow to create public thread from no message, but not discord.py

### What is the feature request for?

The core library

### The Problem

It is impossible to create public thread from a channel using discord.py
Using the API, there is a type parameter, that is selected to PRIVATE in dpy if it is from a channel, and to PUBLIC if it is from a message, but when we start a thread from a channel, the API allow us to create PUBLIC threads.

Like this :
![image](https://user-images.githubusercontent.com/47398145/126567045-0f98ed5e-6da0-4430-bc02-b0d7c1857566.png)

### The Ideal Solution

Add a `type` parameter (or a boolean `private` parameter to the function `discord.TextChannel.start_thread` that allow us to choose if we want to create a PRIVATE thread or a PUBLIC thread.

### The Current Solution

We can make an http request with the public_thread type directly :
```py
        data = await inter.channel._state.http.start_private_thread(
            inter.channel.id,
            name="name"
            auto_archive_duration=1440,
            type=discord.ChannelType.public_thread.value,
        )
        thread = discord.Thread(guild=inter.guild, state=inter.channel._state, data=data)
```
(Using the function `start_private_thread` because `start_public_thread` need a message_id)

**Comment by NCPlayz** [comment]
If the `message` keyword argument of `start_thread` is not provided, it will make a private thread: https://github.com/Rapptz/discord.py/blob/23852404ed86a98e0b83130662368386e76236c9/discord/channel.py#L678-L684

**Comment by AiroPi** [comment]
> If the `message` keyword argument of `start_thread` is not provided, it will make a private thread:
> 
> https://github.com/Rapptz/discord.py/blob/23852404ed86a98e0b83130662368386e76236c9/discord/channel.py#L678-L684

Yes, but the goal is to start a **public** thread from no message, that will display a message like in the screen.

**Comment by NCPlayz** [comment]
Ah! I misread. I understand what you mean.

**Comment by NCPlayz** [comment]
Implemented in dac0267e28ecaef1ba0b25eda25c1ef073ecaeb0
